addons/website_crm/models/crm_lead.py

```diff
@@ -45,9 +45,19 @@ def _merge_get_fields_specific(self):
         return fields_info
 
     def website_form_input_filter(self, request, values):
-        values['medium_id'] = values.get('medium_id') or \
-                              self.sudo().default_get(['medium_id']).get('medium_id') or \
-                              self.sudo().env.ref('utm.utm_medium_website').id
+        try:
+            values['medium_id'] = values.get('medium_id') or \
+                                self.sudo().default_get(['medium_id']).get('medium_id') or \
+                                self.sudo().env.ref('utm.utm_medium_website').id
+        except ValueError:
+            values['medium_id'] = self.sudo().env['utm.medium'].create({'name': 'Website'}).id
+            self.sudo().env['ir.model.data'].create({
+                'name': 'utm_medium_website',
+                'module': 'utm',
+                'res_id': values['medium_id'],
+                'model': 'utm.medium',
+            })
```

Comment:
Would probably have a specific method do that in utm directly, it would probably be useful in other places.

Like `_fetch_or_create_website_medium`